Source/Core/DolphinWX/LuaScriptFrame.cpp

```diff
@@ -0,0 +1,553 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "LuaScripting.h"
+
+#include <algorithm>
+#include <fstream>
+#include <string>
+#include <wx/button.h>
+#include <wx/frame.h>
+#include <wx/menu.h>
+#include <wx/panel.h>
+#include <wx/sizer.h>
+#include <wx/textctrl.h>
+#include "Core\Core.h"
+#include "Core\HW\GCPad.h"
+#include "Core\HW\GCPadEmu.h"
+#include "Core\Movie.h"
+#include "DolphinWX/Main.h"
+#include "DolphinWX/WxUtils.h"
+#include "Frame.h"
+#include "InputCommon\InputConfig.h"
+
+namespace Lua
+{
+int printToTextCtrl(lua_State* L);
+int frameAdvance(lua_State* L);
+int getFrameCount(lua_State* L);
+int getMovieLength(lua_State* L);
+int softReset(lua_State* L);
+
+// Slots from 0-99.
+int saveState(lua_State* L);
+int loadState(lua_State* L);
+
+// The argument should be a percentage of speed. I.E. setEmulatorSpeed(100) would set it to normal
+// playback speed
+int setEmulatorSpeed(lua_State* L);
+
+int getAnalog(lua_State* L);
+int setAnalog(lua_State* L);
+int setAnalogPolar(lua_State* L);
+int getCStick(lua_State* L);
+int setCStick(lua_State* L);
+int setCStickPolar(lua_State* L);
+int getButtons(lua_State* L);
+int setButtons(lua_State* L);
+int setDPad(lua_State* L);
+int getTriggers(lua_State* L);
+int setTriggers(lua_State* L);
+
+// GLOBAL IS NECESSARY FOR LOG TO WORK
+LuaScriptFrame* currentWindow;
+
+// External symbols
+std::map<const char*, LuaFunction>* registered_functions;
+std::mutex lua_mutex;
+
+LuaScriptFrame::LuaScriptFrame(wxWindow* parent)
+    : wxFrame(parent, wxID_ANY, _("Lua Console"), wxDefaultPosition, wxSize(431, 397),
+              wxDEFAULT_FRAME_STYLE ^ wxRESIZE_BORDER)
+{
+  CreateGUI();
+  SetIcons(WxUtils::GetDolphinIconBundle());
+
+  Center();
+  Show();
+
+  currentWindow = this;
+  lua_thread = nullptr;
+
+  // Create function map if it doesn't already exist
+  if (!registered_functions)
+  {
+    registered_functions = new std::map<const char*, LuaFunction>;
+
+    registered_functions->insert(std::pair<const char*, LuaFunction>("print", printToTextCtrl));

```

Comment:
`emplace` can be utilized to simplify this to:

```cpp
...->emplace("print", printToTextCtrl);
```
instead of needing the entry type to be specified before it.